tests/test_fastapi.py

```diff
@@ -63,21 +63,56 @@ def test_protected_endpoint(fastapi_app: FastAPI) -> None:
 def test_protected_with_csrf_endpoint(fastapi_app: FastAPI) -> None:
     with TestClient(fastapi_app) as c:
         # Valid referer header
-        resp = c.get('/protected_with_csrf', headers={"Referer": "http://127.0.0.1"}, cookies={"access_token": client_token['access_token']})
+        resp = c.get('/protected_with_csrf', headers={"Referer": "https://example.com"},
+                     cookies={"access_token": client_token['access_token']})
         assert resp.status_code == 200
         # Invalid referer header
-        resp = c.get('/protected_with_csrf', headers={"Referer": "http://foo.bar"}, cookies={"access_token": client_token['access_token']})
+        resp = c.get('/protected_with_csrf', headers={"Referer": "http://foo.bar"},
+                     cookies={"access_token": client_token['access_token']})
+        assert resp.status_code == 401
+
+
+@iam_mock
+def test_protected_with_csrf_endpoint_with_subdomain(fastapi_app_with_allowing_subdomain: FastAPI) -> None:
+
+    with TestClient(fastapi_app_with_allowing_subdomain) as c:
+        # Valid referer header with subdomain
```

Comment:
You could reuse the fastapi_app fixture here and change the IAM config on this context manager  
`fastapi_app.state.iam.config.iam_allow_subdomain_referer = True`